apps/gdalalg_raster_contour.cpp

```diff
@@ -0,0 +1,221 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster contour" subcommand
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_contour.h"
+
+#include "cpl_conv.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+#include "gdal_alg.h"
+#include "gdal_utils_priv.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*          GDALRasterContourAlgorithm::GDALRasterContourAlgorithm()    */
+/************************************************************************/
+
+GDALRasterContourAlgorithm::GDALRasterContourAlgorithm(
+    bool openForMixedRasterVector)
+    : GDALAlgorithm(NAME, DESCRIPTION, HELP_URL), m_outputLayerName("contour"),
+      m_elevAttributeName(""), m_amin(""), m_amax(""), m_levels{}
+{
+
+    AddProgressArg();
+    AddOutputFormatArg(&m_outputFormat)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES,
+                         {GDAL_DCAP_RASTER, GDAL_DCAP_CREATECOPY});
+    AddOpenOptionsArg(&m_openOptions);
+    AddInputFormatsArg(&m_inputFormats)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES, {GDAL_DCAP_RASTER});
+    AddInputDatasetArg(&m_inputDataset, openForMixedRasterVector
+                                            ? GDAL_OF_RASTER | GDAL_OF_VECTOR |
+                                                  GDAL_OF_MULTIDIM_RASTER
+                                            : GDAL_OF_RASTER);
+    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_RASTER);
```

Comment:
to answer @jratike80 question:
```suggestion
    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_VECTOR);
```